.github/workflows/build-containers.yml

```diff
@@ -0,0 +1,139 @@
+name: Docker smoketest build
+on:
+  pull_request:
+    branches:
+      - "master"
+      - "candidate-*"
+      - "!candidate-8.2.*"
+      - "!candidate-8.0.*"
+      - "!candidate-7.12.*"
+      - "!candidate-7.10.*"
+      - "!candidate-7.8.*"
+      - "!candidate-7.6.*"
+      - "!candidate-7.4.*"
+      - "!candidate-7.2.*"
+      - "!candidate-7.0.*"
+      - "!candidate-6.*"
+
+jobs:
+  check-skip:
+    # continue-on-error: true # Uncomment once integration is finished
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: hpcc-systems/skip-duplicate-actions@master
+        with:
+          github_token: ${{ github.token }}
+          paths_ignore: '["clienttools/**", "devdoc/**", "docs/**", "helm/**", "initfiles/**" ]'
+
+  build-images:
+    needs: check-skip
+    if: ${{ needs.check-skip.outputs.should_skip != 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: vars
+        id: vars
+        run: |
+          echo ::set-output name=base_ver::7.12
+          echo ::set-output name=container_registry::ghcr.io
+          echo ::set-output name=cr_user::${{ github.actor }}
+          echo ::set-output name=build_prbase_sha::${{ github.event.pull_request.base.sha }}
+          echo ::set-output name=build_prbase_label::label-${{ github.event.pull_request.base.sha }}
+          echo ::set-output name=build_pr_label::label-${{ github.sha }}
```

Comment:
It would not make sense for the base image to contain the PR # , since the base image will be used by many different PR's. I can add the PR-nnnn to the build_pr_label, although as there is no current intention to push those images, meaning this label is purely local to this action, it makes little difference.

I've changed the base image tag, so it contains the base_ref branch name.
I've also change PR image so it contains PR-nnnn (even though not pushed)